assets/ajax-reload.js

```diff
@@ -0,0 +1,142 @@
+window.HasteAjaxReload = {
+    elementsInProgress: {},
+    eventsInProgress: {},
+
+    /**
+     * Dispatch the events
+     */
+    dispatchEvents: function () {
+        if (arguments.length < 1) {
+            console.error('Please provide at least one event');
+        }
+
+        var els = {};
+        var events = [];
+        var listeners = document.querySelectorAll('[data-haste-ajax-listeners]');
+
+        Array.from(arguments).forEach(function (event) {
+            var found = false;
+
+            // Find the elements that listen to particular event
+            Array.from(listeners).forEach(function (el) {
+                if (el.dataset.hasteAjaxListeners.split(' ').indexOf(event) !== -1) {
+                    found = true;
+                    els[el.dataset.hasteAjaxId] = el;
+                }
+            });
+
+            if (!found) {
+                console.warn('There are no eligible elements for event "' + event + '"');
+            } else {
+                events.push(event);
+            }
+        });
+
+        if (Object.keys(els).length > 0 && events.length > 0) {
+            this._sendRequest(els, events);
+        }
+    },
+
+    /**
+     * Send the request
+     * @param {Object} els
+     * @param {Array} events
+     * @private
+     */
+    _sendRequest: function (els, events) {
```

Comment:
They should not but I want to keep everything in the same object/class and since there is no real privacy for methods in the JS (nor in PHP actually)  I just added the commonly used fake-privacy stuff.